test/client_certificate_test.dart

```diff
@@ -0,0 +1,131 @@
+@TestOn('vm')
+import 'dart:async';
+import 'dart:io';
+
+import 'package:grpc/grpc.dart';
+import 'package:test/test.dart';
+
+import 'src/generated/echo.pbgrpc.dart';
+
+class EchoService extends EchoServiceBase {
+  @override
+  Future<EchoResponse> echo(ServiceCall call, EchoRequest request) async {
+    final subject = call.clientCertificate?.subject;
+    return (EchoResponse()..message = subject ?? 'NO CERT');
+  }
+
+  @override
+  Stream<ServerStreamingEchoResponse> serverStreamingEcho(
+      ServiceCall call, ServerStreamingEchoRequest request) {
+    // TODO: implement serverStreamingEcho
+    throw UnimplementedError();
+  }
+}
+
+const String address = 'localhost';
+const int port = 19360;
```

Comment:
You should avoid hardcoding the port, because the test can easily fail if the port is not available. Instead you can ask `server.port` after you start the server.